src/modules.cpp

```diff
@@ -838,6 +838,23 @@ PVOID WINAPI DetourFindPayloadEx(_In_ REFGUID rguid,
     return NULL;
 }
 
+BOOL WINAPI DetourFreePayload(_In_ PVOID pvData)
+{
+    BOOL fSucceeded = FALSE;
+
+    HMODULE hModule = DetourGetContainingModule(pvData);
```

Comment:
I think this deserves a comment describing that you are (cleverly) using `DetourGetContainingModule(..)` to get the root of the `VirtualAlloc` that was created in `DetourCopyPayloadToProcess(..)`. Otherwise it's going to take the reader a bit to figure out what's going on here, and potentially looking like a bug.

Reply:
Good idea, the latest submission has been changed in accordance with your suggestions.